src/hooks/useSortedAndSearchedProducts.js

```diff
@@ -0,0 +1,53 @@
+import * as React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { getProductList } from "../store/productSlice";
+
+const reducer = (state, action) => {
+    switch (action.type) {
+        case "SET_SORT_BY":
+            return { ...state, sortBy: action.sortBy };
+        case "SET_SEARCH":
+            return { ...state, search: action.search };
+        case "SET_JUST_MOUNTED":
+            return { ...state, justMounted: false };
+        default:
+            return state;
+    }
+};
+
+const useSortedAndSearchedProducts = (initialSortBy = "CREATED_TIME", initialSearch = "") => {
+    const { isLoading, byIds, ids, isFirstLoad } = useSelector((state) => state.product);
+    const [state, setState] = React.useReducer(reducer, {
+        sortBy: initialSortBy,
+        search: initialSearch,
+        justMounted: true,
+    });
+    const productList = React.useMemo(() => ids.map((id) => byIds[id]), [ids, byIds]);
+
+    console.log("in custom hook");
+    const { sortBy, search, justMounted } = state;
+    const dispatch = useDispatch();
+
+    const setSortBy = React.useCallback((sortBy) => {
+        setState({ type: "SET_SORT_BY", sortBy });
```

Comment:
Should create constant for "SET_SORT_BY", "SET_SEARCH", "SET_JUST_MOUNTED", "CREATED_TIME"